composer.json

```diff
@@ -38,7 +37,10 @@
         "zendframework/zendframework" : "2.5.3",
         "particle/validator": "2.3.3",
         "guzzlehttp/guzzle": "6.3.3",
-        "kamermans/guzzle-oauth2-subscriber": "1.0.5"
+        "kamermans/guzzle-oauth2-subscriber": "1.0.5",
+        "league/omnipay": "3.0.1",
+        "academe/omnipay-authorizenetapi": "3.0.4",
+        "omnipay/stripe": "3.0.1"
```

Comment:
You may need to update composer to get these changes in, which will update the composer.lock file. Try:
composer update
composer dump-autoload -o
(this should then show the composer.lock as being changed which will then be part of your commit/branch; btw, this is the file that is basically used when running composer and ensures we are all using the same version of packages)

Reply:
While thinking about including the lock file in master i'm not sure that is a good idea. Though we want to ensure a set version for our dependencies it is too easy to get the committed lock version out of sync with the composer jason version.  We already lock to the desired version in the composer json so the generated locked file should mirror the fetched dependency version. Case in point, because I didn't push a lock file, my dependencies were not included on the demo. If I included the lock file from my dev branch but forgot to remove dependencies I added for development on project then lock is out of sync with my committed composer json.

Reply:
we do pin versions, however there are lots of dependencies of those versions that keep changing. committing the composer.lock is recommended, however, as with everything else there is controversy: https://stackoverflow.com/questions/12896780/should-composer-lock-be-committed-to-version-control
(my specific use care concern with not including the composer.lock is that when we do the final production build for a version release, that it will bring in a upgraded sub-dependency that breaks everything)

Reply:
I don't know. Too me it just turns into a circle argument. Like the joke; it hurts when I do this... well don't do that. Either way, there are pitfalls and I suppose maintaining the lock as dev moves forward will at least give you a look at what dependencies change as we go. For production, i'd bet 9 out of 10 are going to run update:)